.github/workflows/build-website.yml

```diff
@@ -11,64 +11,92 @@ jobs:
   build:
     name: website
     runs-on: ubuntu-latest
-    # TODO(strager): Use an image based on
-    # quick-lint-js-web.
-    container: ghcr.io/quick-lint/quick-lint-js-github-builder:v3
+    container: ghcr.io/quick-lint/quick-lint-js-website-builder:v1
     env:
       QLJS_COLLECT_COPYRIGHT_NO_WARNINGS: 1
+      EM_VERSION: 2.0.4
+      EM_CACHE_FOLDER: 'emsdk-cache'
+
     steps:
-      - uses: quick-lint/setup-emsdk@v7
-        with:
-          version: 2.0.4
       - name: checkout
         uses: actions/checkout@v2
 
+      - name: setup emsdk cache
+        id: cache-system-libraries
+        uses: actions/cache@v2
+        with:
+          path: ${{env.EM_CACHE_FOLDER}}
+          key: ${{env.EM_VERSION}}-${{ runner.os }}
+
+      - uses: quick-lint/setup-emsdk@v7
+        with:
+          version: ${{env.EM_VERSION}}
+          actions-cache-folder: ${{env.EM_CACHE_FOLDER}}
+
       - name: C++ configure
-        run: emcmake cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS='-Os -flto' -DCMAKE_CXX_FLAGS='-Os -flto'
+        run: emcmake cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=NO -DCMAKE_C_FLAGS='-Os -flto' -DCMAKE_CXX_FLAGS='-Os -flto'
       - name: C++ build
         run: emmake cmake --build build --target quick-lint-js-vscode quick-lint-js-vscode-licenses
       - name: C++ install
         run: emmake cmake --install build --component vscode --prefix website/public/demo
+      
+      - name: get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v1
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
```

Comment:
How does `restore-keys` work in this case? From what I can tell, this will basically accomplish nothing. If that's true, then why specify `restore-keys` at all?